### ISSUE TITLE
User does not exists when log opened

### ISSUE THREAD

**Issue by striderxfossility** [opcodesio/log-viewer#264]
I have this as AuthServiceProvider:
```
Gate::define('viewLogViewer', function (?User $user) {
      if($user === null)
          return false;

      if ($user->has_role('super-admin'))
          return true;

      return false;
  });
```
        
       This works, but when i am on the page itself it says: Request failed with status code 403: This action is unauthorized.

its right under the title LogViewer

**Comment by arukompas** [comment]
Hey @striderxfossility , could you dump the `$user` variable just before any if statements?

Second, do you have the stateful middleware applied to the `'api_middleware'` config option? It should be there by default if you're using a new Log Viewer install.

And third, are you using the default cookie-based sessions, or a different method for authenticating your users?

**Comment by striderxfossility** [comment]
Yes here is dump($user) in that gate

[07:52:04] LOG.info: {"id":1,"name":"Jelle Bakker","email":"****","email_verified_at":"2023-08-19T17:57:52.000000Z","created_at":"2023-08-19T17:57:52.000000Z","updated_at":"2023-08-19T17:57:52.000000Z","deleted_at":null,"permissions":[],"roles":[{"id":1,"name":"super-admin","guard_name":"super-admin","created_at":"2023-08-19T17:57:52.000000Z","updated_at":"2023-08-19T17:57:52.000000Z","description":"Super Administrator","pivot":{"model_id":1,"role_id":1,"model_type":"App\\Models\\User"},"permissions":[]}]}

Yes this is the config:
```
'middleware' => [
        'web',
        \Opcodes\LogViewer\Http\Middleware\AuthorizeLogViewer::class,
],
'api_middleware' => [
        \Opcodes\LogViewer\Http\Middleware\EnsureFrontendRequestsAreStateful::class,
        \Opcodes\LogViewer\Http\Middleware\AuthorizeLogViewer::class,
],
```

i also tried 'middelware' => ['role:super-admin'] but that did not work.

i am using the following:
"php": "^8.2",
"guzzlehttp/guzzle": "^7.2",
"laravel/breeze": "^1.23",
"laravel/fortify": "^1.17",
"laravel/framework": "^10.10",
"laravel/passport": "^11.8",
"laravel/sanctum": "^3.2",
"laravel/tinker": "^2.8",
"opcodesio/log-viewer": "^3.0",
"spatie/laravel-permission": "^5.10"

**Comment by arukompas** [comment]
@striderxfossility
Ok, so you see the user does get resolved here, so it's not null.

My guess is that there's something wrong with the role check, and then it just returns false. Double-check that first, maybe dump the result of the role check (whether it's true or false)

**Comment by striderxfossility** [comment]
Well it works, i see the log-viewer url when i am super admin, and not if i am admin.

but its inside the view that is not working.

[Back to Laravel](http://localhost/)
Request failed with status code 403: This action is unauthorized.

No log files were found. i get this message inside of /log-viewer when i am authorized to view it.

so i see your layout view when i am super admin.

**Comment by arukompas** [comment]
hey @striderxfossility , I assume you've used Spatie permissions to assign roles. The `$user->has_role()` method is not available, so I guess that's a custom method you defined?

Either way, I've spend time re-creating the issue, installing Laravel Breeze (blade), then Spatie Permissions package, setting up the `super-admin` role, and it was working as expected. 

I tried both with Gate definition:
```php
        Gate::define('viewLogViewer', function (?User $user) {
            if ($user === null) {
                return false;
            }

            if ($user->hasRole('super-admin')) {
                return true;
            }

            return false;
        });

```

And also by adding the role middleware in `config/log-viewer.php`:
```php
    'middleware' => [
        'web',
        // \Opcodes\LogViewer\Http\Middleware\AuthorizeLogViewer::class,
        'role:super-admin',
    ],

    'middleware' => [
        'web',
        // \Opcodes\LogViewer\Http\Middleware\AuthorizeLogViewer::class,
        'role:super-admin',
    ],
```

and both options work as expected.

If you have a public git repository where this bug can be reproduced, I'd be happy to look at it. Otherwise, it seems like an issue might be elsewhere.

I also noticed the guard name for your role is `super-admin` as well, instead of the default `web`. How did you set up your auth guards, and why? Why aren't you using the default `web` guard for authenticating this user?
